Framework/Kernel/inc/MantidKernel/InstrumentInfo.h

```diff
@@ -96,10 +104,11 @@ class MANTID_KERNEL_DLL InstrumentInfo {
   std::string m_shortName;        ///< Instrument short name
   ZeroPaddingMap m_zeroPadding;   ///< Run number-dependent zero padding
   std::string m_delimiter; ///< Delimiter between instrument name and run number
-  std::string m_liveListener;    ///< Name of the live listener class
-  std::string m_liveDataAddress; ///< Host & port for live data connection
   std::set<std::string>
       m_technique; ///< List of techniques the instrument can do
+
+  std::vector<LiveListenerInfo> m_listeners; ///< LiveListener connections
```

Comment:
I gave it a try, but got a bunch of compiler errors. It seems that std::vector elements must be assignable.